File: bulk_enroll.js
const Database = require('better-sqlite3');
const path = require('path');

const dbPath = path.join(process.env.APPDATA, 'school-management-system', 'school.db');
const db = new Database(dbPath);

const students = db.prepare("SELECT fullName, className, photo FROM students WHERE photo IS NOT NULL AND photo != ''").all();

async function processStudents() {
  console.log(`Found ${students.length} students with photos.`);
  let successCount = 0;
  let failCount = 0;
  
  for (const student of students) {
    if (!student.photo.startsWith('data:image')) {
      console.log(`Skipping ${student.fullName}, photo is not base64.`);
      continue;
    }
    
    try {
      const base64Data = student.photo.replace(/^data:image\/\w+;base64,/, "");
      const buffer = Buffer.from(base64Data, 'base64');
      const blob = new Blob([buffer], { type: 'image/jpeg' });
      const formData = new FormData();
      formData.append('file', blob, 'face.jpg');
      
      const response = await fetch('http://127.0.0.1:5000/extract', {
        method: 'POST',
        body: formData
      });
      
      if (!response.ok) {
        console.log(`❌ HTTP Error for ${student.fullName}`);
        failCount++;
        continue;
      }
      
      const data = await response.json();
      if (data.faces && data.faces.length > 0) {
        const embedding = data.faces[0].embedding;
        db.prepare(`DELETE FROM faceEmbeddings WHERE targetType='students' AND name=?`).run(student.fullName);
        db.prepare(`
          INSERT INTO faceEmbeddings (targetType, name, tag, descriptorJson)
          VALUES ('students', ?, ?, ?)
        `).run(student.fullName, student.className || '', JSON.stringify(embedding));
        
        console.log(`✅ Enrolled ${student.fullName}`);
        successCount++;
      } else {
        console.log(`❌ No face detected for ${student.fullName}`);
        failCount++;
      }
    } catch (err) {
      console.log(`Error processing ${student.fullName}: ${err.message}`);
    }
  }
  
  console.log(`\nFinished! Enrolled: ${successCount}, Failed/No Face: ${failCount}`);
}

processStudents();

File: bulk_enroll.py
import sqlite3
import os
import json
import base64
import urllib.request
import urllib.parse
import binascii

db_path = os.path.join(os.path.expanduser('~'), 'AppData', 'Roaming', 'school-management-system', 'school.db')
conn = sqlite3.connect(db_path)
cursor = conn.cursor()

cursor.execute("SELECT fullName, className, photo FROM students WHERE photo IS NOT NULL AND photo != ''")
students = cursor.fetchall()

print(f"Found {len(students)} students with photos.")
success_count = 0
fail_count = 0

for student in students:
    full_name, class_name, photo = student
    if not photo.startswith('data:image'):
        print(f"Skipping {full_name}, photo is not base64.")
        continue
    
    print(f"Processing {full_name}...")
    try:
        header, encoded = photo.split(',', 1)
        image_data = base64.b64decode(encoded)
        
        boundary = '----WebKitFormBoundary7MA4YWxkTrZu0gW'
        body = b'--' + boundary.encode() + b'\r\n'
        body += b'Content-Disposition: form-data; name="file"; filename="photo.jpg"\r\n'
        body += b'Content-Type: image/jpeg\r\n\r\n'
        body += image_data
        body += b'\r\n--' + boundary.encode() + b'--\r\n'
        
        req = urllib.request.Request('http://127.0.0.1:8000/extract', data=body)
        req.add_header('Content-Type', f'multipart/form-data; boundary={boundary}')
        req.add_header('Content-Length', str(len(body)))
        
        with urllib.request.urlopen(req) as response:
            if response.status != 200:
                print(f"HTTP Error for {full_name}: {response.status}")
                fail_count += 1
                continue
            
            data = json.loads(response.read().decode('utf-8'))
            if data.get('faces') and len(data['faces']) > 0:
                embedding = data['faces'][0]['embedding']
                
                cursor.execute("DELETE FROM faceEmbeddings WHERE targetType='students' AND name=?", (full_name,))
                cursor.execute("""
                    INSERT INTO faceEmbeddings (targetType, name, tag, descriptorJson)
                    VALUES ('students', ?, ?, ?)
                """, (full_name, class_name or '', json.dumps(embedding)))
                
                print(f"Enrolled {full_name}")
                success_count += 1
            else:
                print(f"No face detected for {full_name}")
                fail_count += 1
                
    except Exception as e:
        print(f"Error processing {full_name}: {e}")

conn.commit()
conn.close()

print(f"\nFinished! Enrolled: {success_count}, Failed/No Face: {fail_count}")
